src/events.rs

```diff
@@ -41,14 +41,52 @@ impl From<&EventModifierFlag> for NSUInteger {
             EventModifierFlag::Control => 1 << 18,
             EventModifierFlag::Option => 1 << 19,
             EventModifierFlag::Command => 1 << 20,
-            EventModifierFlag::DeviceIndependentFlagsMask => 0xffff0000
+            EventModifierFlag::DeviceIndependentFlagsMask => 0xffff0000,
         }
     }
 }
 
 /// Represents an event type that you can request to be notified about.
 #[derive(Clone, Copy, Debug)]
+#[cfg_attr(target_pointer_width = "32", repr(u32))]
+#[cfg_attr(target_pointer_width = "64", repr(u64))]
 pub enum EventType {
-    /// A keydown event.
-    KeyDown
+    LeftMouseDown = 1,
+    LeftMouseUp = 2,
+    RightMouseDown = 3,
+    RightMouseUp = 4,
+    MouseMoved = 5,
+    LeftMouseDragged = 6,
+    RightMouseDragged = 7,
+    MouseEntered = 8,
+    MouseExited = 9,
+    KeyDown = 10,
+    KeyUp = 11,
+    FlagsChanged = 12,
+    AppKitDefined = 13,
+    SystemDefined = 14,
+    ApplicationDefined = 15,
+    Periodic = 16,
+    CursorUpdate = 17,
+
+    ScrollWheel = 22,
+    TabletPoint = 23,
+    TabletProximity = 24,
+    OtherMouseDown = 25,
+    OtherMouseUp = 26,
+    OtherMouseDragged = 27,
+
+    Gesture = 29,
+    Magnify = 30,
+    Swipe = 31,
+    Rotate = 18,
+    BeginGesture = 19,
+    EndGesture = 20,
+
+    SmartMagnify = 32,
+    QuickLook = 33,
+    Pressure = 34,
+    DirectTouch = 37,
+
+    ChangeMode = 38,
```

Comment:
It would be really nice if these shift values didn't have to be duplicated, though I don't think there's a nice way to do it. Apple assumedly has a set of `k*` constants that feed both enums.

Reply:
Hmm, one thing I experimented with in the `color` module was this:

https://github.com/ryanmcgrath/cacao/blob/9f8d946371f4a6dba7346052ca38011a101d3beb/src/color/appkit_dynamic_color.rs#L27-L33

Maybe there's a similar approach here? e.g somehow just pulling in the already linked `NSEventMaskLeftMouseDown` variant.

Reply:
I don't think that would work, as enums need `const`s (basically `#define`) to set their values, so we would need to pull out the constant from the header. It works that way in your example because those strings aren't true constants.

Reply:
Yeah... you're right about that - was a late night last night so disregard my rambling thoughts. I thought I had encountered and found a workaround for this some years ago but guess I'm misremembering.

This is probably good in its current result to merge then, unless you've got anything else you can think of?

Reply:
This is fine by me for now.